packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -251,6 +254,34 @@ export abstract class AbstractBackgroundJobProcessor<
 		return jobIds as string[]
 	}
 
+	/**
+	 * Get jobs in the given states.
+	 *
+	 * @param states
+	 * @param start default 0
+	 * @param end default 20
+	 * @param asc default true (oldest first)
+	 */
+	public async getJobsInQueue(
+		states: JobState[],
+		start: number = 0,
+		end: number = 20,
+		asc: boolean = true,
+	): Promise<JobsPaginatedResponse<JobPayload, JobReturn>> {
+		if (states.length === 0) throw new Error('states must not be empty')
+		if (start > end) throw new Error('start must be less than or equal to end')
+
+		await this.startIfNotStarted()
+
+		const jobs = (await this.queue?.getJobs(states, start, end + 1, asc)) ?? []
```

Comment:
I would say that is safe to get more jobs from the queue, but I wanted to give some default values to avoid misusage and return a huge amount of them and decided to choose a low value allowing users to choose the right values they want to use